components/ILIAS/Authentication/classes/Logout/ConfigurableLogoutHandler.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/**
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ *
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ *
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *
+ *********************************************************************/
+
+declare(strict_types=1);
+
+namespace ILIAS\components\Authentication\Logout;
+
+use ilLink;
+use ilSetting;
+use ilStartUpGUI;
+use ILIAS\Data\URI;
+use ilCtrlInterface;
+
+class ConfigurableLogoutHandler
```

Comment:
We should do some brainstorming for a proper naming. IMO this is not a "Handler", because this class does not handle something, consumers can retrieve an URL based on a given configuration.